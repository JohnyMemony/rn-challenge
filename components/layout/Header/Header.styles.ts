import { StyleSheet } from 'react-native';
import Colors from '../../../constants/Colors';
import { screenFreeSpace } from '../../../constants/Layout';

export const styles = StyleSheet.create({
  wrap: {
    height: 100,
    paddingTop: 40,
    paddingBottom: 13,
    paddingHorizontal: screenFreeSpace,
  },
  row: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  title: {
    fontSize: 19,
    fontWeight: '700',
    color: Colors.white,
  },
});
