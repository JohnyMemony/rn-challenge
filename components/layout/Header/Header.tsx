import React from 'react';
import { Text, View } from 'react-native';
import { styles } from './Header.styles';

interface Props {
  title: string;
  actions: React.ReactNode;
}

export const Header: React.FC<Props> = (props) => {
  const { title, actions } = props;

  return (
    <View style={styles.wrap}>
      <View style={styles.row}>
        <Text style={styles.title}>{title}</Text>
        <>{actions}</>
      </View>
    </View>
  );
};
