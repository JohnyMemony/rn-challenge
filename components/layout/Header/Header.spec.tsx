import React from 'react';
import renderer from 'react-test-renderer';
import { Header } from './Header';

test('renders header correctly', () => {
  const tree = renderer.create(<Header title="Test" actions={(<></>)} />).toJSON();

  expect(tree).toMatchSnapshot();
});
