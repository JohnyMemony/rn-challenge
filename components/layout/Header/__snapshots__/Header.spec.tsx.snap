// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header correctly 1`] = `
<View
  style={
    Object {
      "height": 100,
      "paddingBottom": 13,
      "paddingHorizontal": 15,
      "paddingTop": 40,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 19,
          "fontWeight": "700",
        }
      }
    >
      Test
    </Text>
  </View>
</View>
`;
